velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -421,20 +422,45 @@ std::optional<TimestampUnit> getTimestampUnit(
   return std::nullopt;
 }
 
+std::optional<std::string> getParquetDataPageVersion(
+    const Config& config,
+    const char* configKey) {
+  const auto version = config.get<std::string>(configKey);
+  VELOX_CHECK(
+      version == "PARQUET_1_0" || version == "PARQUET_2_0",
+      "Invalid Parquet version.");
+  return version.value();
+}
+
 } // namespace
 
 void WriterOptions::processSessionConfigs(const Config& config) {
   if (!parquetWriteTimestampUnit) {
     parquetWriteTimestampUnit =
         getTimestampUnit(config, kParquetSessionWriteTimestampUnit);
   }
+
+  auto parquetDataPageVersionString =
+      getParquetDataPageVersion(config, kParquetSessionDataPageVersion);
+
+  if (parquetDataPageVersionString == "PARQUET_1_0") {
```

Comment:
maybe having a helper method with a switch to do the mapping, and throwing in case you get an invalid value instead of defaulting to v2?

```
parquetDataPageVersionString = toParquetDataPage(config);
```

Reply:
I see what you mean, but also realized I could just be doing the mapping in `getParquetDataPageVersion()` in combination with validation, and directly return a `arrow::ParquetDataPageVersion` value. Let me know if that looks ok.